### ISSUE TITLE
Using HAXEPATH Environment Variable

### ISSUE THREAD

**Issue by drkibitz** [HaxeFoundation/haxelib#100]
This issue is in regards to the HAXEPATH environment variable, and the specificity it has to Windows. This should not be the case, the logic used in checking the value should be the same for all platforms. Would be very helpful for any other more complex environments.

**Comment by back2dos** [comment]
Can you elaborate on this one? I think HAXEPATH really is a windows only thing, while HAXELIB_PATH is the cross platform way to set the path through environment variables.

**Comment by ibilon** [comment]
Maybe a reference to this https://github.com/HaxeFoundation/haxelib/blob/master/src/tools/haxelib/Main.hx#L980-L994
The fact that HAXEPATH is used as a default value if HAXELIB_PATH isn't set and there's no .haxelib in the user's home directory but only on windows.

**Comment by andyli** [comment]
Let's reopen this one to see if we can remove the use of `HAXEPATH` even on Windows :)
See https://github.com/HaxeFoundation/haxelib/commit/bd5740ca589287acc719c62a5e9dd0049ce1b905#commitcomment-14917006

**Comment by ncannasse** [comment]
HAXEPATH is set by Windows installer. We don't want users on Windows to have to run haxelib setup before being able to install libraries.

**Comment by nadako** [comment]
Yes, but can't we achieve this by determining haxelib.exe folder somehow and using that instead of HAXEPATH?

**Comment by ncannasse** [comment]
That's usually how I would do :) So if at some point HAXEPATH was used someone must have had a problem with it. Can't tell unless we look at full history

**Comment by nadako** [comment]
I think we can try using `Sys.executablePath()` on Windows, since we're always dealing with `haxelib.exe` there. Unless we want to change it to some `haxelib.cmd` batch file that does `--run`
